### ISSUE TITLE
Product Summary: HTML is stripped out and trimmed in the short description

### ISSUE THREAD

**Issue by a3rev** [woothemes/woocommerce#44245]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

WooCommerce single product block template, Product Summary Block,  strips out all HTML from short descriptions eg p tags and trims the content when rendered on fronted. As reported [here on wordpress.org](https://wordpress.org/support/topic/post-full-short-description-in-product-page/)

Asked by @shameemreza on that support thread to open this issue.

Short description shows as a single paragraph block of text. We have added this custom code to the child themes functions.php to 'fix' the issue. This is currently working on 7 production sites, so it is well tested.

`add_filter('wp_trim_words', function ($text, $num_words, $more, $original_text) {
    if( is_singular( 'product' ) ){
        $text = wpautop( $original_text );
        return $text;
    }
}, 10, 4);`

The issue has been tested and replicated by WordPress user (woo-hc) @carolm29

### Expected behavior

Product short description should support editor HTML tags - example p tag. The tags are supported in the editor, but stripped out on the frontend display.

### Actual behavior

All tags are stripped out and the content length is trimmed on frontend display.

### Steps to reproduce

1. Install Default WordPress theme and WooCommerce.
2. Create a Product. Write a short description with paragraphs, bold text, bullet points etc,
3. Publish the product and view the frontend
4. You will see all HTML tags have been stripped and the Short description has been trimmed   

### WordPress Environment

Tested on the latest WordPress and WooCommerce versions.

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by shameemreza** [comment]
Related to this: https://github.com/woocommerce/woocommerce/issues/44185

**Comment by kmanijak** [comment]
Thanks for raising this!

I can see two separate issues raised in this issue. Let me respond to them separately:

1. **Product Summary is truncated** - that is an expected result, however, it's not well exposed in the Product Editor and creates confusion. There's ongoing discussion: https://github.com/woocommerce/woocommerce/discussions/44260 in which we discuss how to address it. Please follow this discussion for further updates on this and feel free to take part in it as well!

2. **Product Summary doesn't support tags formatting** - I can reproduce it and agree it's an issue, especially, that classic Single Product template had supported that behavior.

I'm allowing myself to change the issue title to highlight it's mainly about the "second" issue and move it to the backlog.

Thanks for raising it!

---

**Tech notes:**
- There's a similar issue in Gutenberg about the post-excerpt block: https://github.com/WordPress/gutenberg/issues/55562.
- Product Editor explicitly supports formatting with WYSWIG Editor so should be displayed in such state as well.

**Comment by filipanoscampos** [comment]
Another case here 7917680-zd-a8c

**Comment by sverleis** [comment]
Customer experiencing this: 8009310-zen

**Comment by aprilfast** [comment]
Hi, just checking in on the status of this. I'm not really sure how to use GitHub correctly, so I hope I'm not being a pest.

**Comment by kmanijak** [comment]
Hi @aprilfast and others! I took at Product Summary recently and I summarised my findings and actions in the discussion we have opened in parallel. Here's the latest update from my side: https://github.com/woocommerce/woocommerce/discussions/44260#discussioncomment-9699418

**TLDR:**
- I increased the default excerpt length to 100 words (maximum allowed by Excerpt block) (https://github.com/woocommerce/woocommerce/pull/47651)
- I contributed to Gutenberg so the length we set is respected if the excerpt is not available (https://github.com/WordPress/gutenberg/pull/61888)
- But to fully solve this problem we need to create a dedicated block (or adopt one of the existing ones) while for now Product Summary is a variation of Gutenberg's Excerpt and clearly there's a clash of requirements.

I'll give it another shot in 2-3 weeks.

**Comment by aprilfast** [comment]
I've written a function to display a shortcode instead, and I'll use that as a stopgap solution until this gets sorted out. I understand that it takes time, but I can't run an ecommerce website that can't display Short Descriptions. Thank you for your work - I sincerely appreciate it.

Editing to add, if anyone wants the snippet, lemme know? I don't think this is the place for it. Again, not really sure how GitHub protocol works.

**Comment by lizswafford** [comment]
Similar issue with formatting here 8477639-zd-a8c

**Comment by ResponsiveFlow** [comment]
hey @kmanijak do you have any updates on this issue?

**Comment by kmanijak** [comment]
@tomxygen, actually I do. Here's the PR: https://github.com/woocommerce/woocommerce/pull/48984 in which I explored making the Product Summary block:
- keeping the styling from the Product Editor
- allowing not to trim the description
- allowing to disable fallback to full product description.

It requires some polishing and writing tests. I hope to finalise it for WooCommerce 9.3.

In the meantime you can test and play around with this block [in WordPress Playground](https://playground.wordpress.net/#%7B%22landingPage%22:%22/wp-admin/admin.php?page=wc-admin%22,%22preferredVersions%22:%7B%22php%22:%228.0%22,%22wp%22:%22latest%22%7D,%22phpExtensionBundles%22:%5B%22kitchen-sink%22%5D,%22features%22:%7B%22networking%22:true%7D,%22steps%22:%5B%7B%22step%22:%22installPlugin%22,%22pluginZipFile%22:%7B%22resource%22:%22url%22,%22url%22:%22https://playground.wordpress.net/plugin-proxy.php?org=woocommerce&repo=woocommerce&workflow=Build%20Live%20Branch&artifact=plugins-9809311478&pr=48984%22%7D,%22options%22:%7B%22activate%22:true%7D%7D,%7B%22step%22:%22installPlugin%22,%22pluginZipFile%22:%7B%22resource%22:%22url%22,%22url%22:%22https://github-proxy.com/https://github.com/woocommerce/woocommerce/releases/download/wc-beta-tester-2.3.1/woocommerce-beta-tester.zip%22%7D,%22options%22:%7B%22activate%22:true%7D%7D,%7B%22step%22:%22login%22,%22username%22:%22admin%22,%22password%22:%22password%22%7D%5D,%22plugins%22:%5B%5D%7D). Let me know if that works as you'd expect! 🙌

**Comment by mrfoxtalbot** [comment]
I just ran into this issue in 8544666-zd-a8c

**Comment by ccwalburn** [comment]
Another report here: 8513980-zd-a8c

Suggested that they could replace the Product Summary block with the Product Details block in the Single Product template as a workaround.

**Comment by addich** [comment]
I have a page which use Thai and there it trimm the short description to 100 characters, which makes it quite unusable.

The reason is that Thai language does not need spaces as we use in English.
Wordpress  has a setting per locale how to count words. It can be words, characters_excluding_spaces or characters_including_spaces. For Thai it is characters_including_spaces.

https://make.wordpress.org/polyglots/2022/09/07/word-counting-type-setting-for-locales/
 
some more background about thai language:
https://thaitranslation.wordpress.com/2008/09/24/thai-and-the-problem-of-word-counts/

So for this kind of Languages it would be great to disable the trim or have a much higher limit then 100 words/charachters.

**Comment by kmanijak** [comment]
Hi folks, I've got an update for you! https://github.com/woocommerce/woocommerce/pull/48984 has been merged and will land in WooCommerce 9.6. 

**Changes include:**
- Providing improvements to the existing Product Summary block (from Product Elements):
    - support for text styling from Product Editor,
    - ability to turn on/off truncation (0 - 200 words, where 0 means no truncation),
    - ability to disable fallback to full product description,
    - "Read more" link,
    - making it compatible and working in different contexts: All Products, Single Product block, Single Product template, Product Collection.
    - adjusting compatibility layer so it takes new block into account
    - add support for languages with word definition as `characters_excluding_spaces` and `characters_including_spaces` (only on the frontend, Editor has the right implementation but the setting doesn't appear in wcBlocksSettings. It requires additional investigation: https://github.com/woocommerce/woocommerce/issues/53146)

**How to use a new block:**
- Block will be available in the inserter under "Product Summary" name
- If you're adding new blocks, the new Product Summary will be already in the markup of:
    - Single Product block
    - when transforming Classic Single Product Template into blockified version
- We don't automatically upgrade the current block (Post Excerpt's variation). So your blocks in use will work as-is but we're providing an upgrade path from current variation(s) to the new Product Summary block through Inspector Controls:

![Image](https://github.com/user-attachments/assets/239888d5-d988-4baa-985f-2d30367d24aa)

- The current Product Summary is NOT replaced in blockified Single Product template. This is a conscious decision since that would affect existing stores with a pristine Single Product template. Product Summary would look different out of nowhere and could be perceived as breaking change. If you'd like to use the new block here, replace it manually or use the upgrade path presented above.

---

Thanks all for you patience. The PR is merged into `trunk` so the block is ready for testing in current `trunk` or will be available in upcoming Nightly build! Any feedback is welcome! 🙌

**Comment by InternetPresence** [comment]
Thank you, that worked beautifully.
